src/search.cpp

```diff
@@ -243,6 +244,23 @@ void MainThread::search() {
   DrawValue[ us] = VALUE_DRAW - Value(contempt);
   DrawValue[~us] = VALUE_DRAW + Value(contempt);
 
+  // Set TB probing parameters
+  TB::DrawScore = Options["Syzygy50MoveRule"] ? TB::WDLCursedWin : TB::WDLDraw;
+  TB::ProbeDepth = Options["SyzygyProbeDepth"] * ONE_PLY;
+  TB::Cardinality = Options["SyzygyProbeLimit"];
+
+  if (TB::Cardinality > TB::MaxCardinality)
+  {
+      TB::Cardinality = TB::MaxCardinality;
+      TB::ProbeDepth = DEPTH_ZERO; // Hack!
+  }
+
+  // Enable probing of DTZ only when WDL is not enough
+  TB::ProbeDTZ =   TB::RootPosDTZ != TB::DTZ_NONE // Only if root is in TB
+                && TB::DrawScore != TB::WDLDraw   // Only if rule50 is enabled
+                && TB::RootPosDTZ                 // Not a draw
+                && rootPos.rule50_count() > 0;    // Not after a resetting move
```

Comment:
I think this line introduces a bug. Although it is true that you can get accurate WDL information for the root position after a zeroing move you need DTZ information in order to accurately evaluate the positions at ply 1. This is crucial in positions that are right on the edge of being cursed/blessed.

A specific counterexample is the KNNKP position 8/4k3/8/3K4/4pN2/4N3/8/8 b - - which is a blessed loss with DTZ=101. At depth 60 stockfish suggests the inferior move Kd7 which brings us into a firmly lost position. The drawing moves are Kf6 and Kf7.

```uci
setoption name SyzygyPath value <PATH>
position fen 8/4k3/8/3K4/4pN2/4N3/8/8 b - -
go depth 60
...
bestmove e7d7 ponder f4g6
```